styx-api-service/src/main/java/com/spotify/styx/api/SchedulerProxyResource.java

```diff
@@ -73,7 +75,20 @@ public SchedulerProxyResource(String schedulerServiceBaseUrl, Client client) {
   }
 
   private CompletionStage<Response<ByteString>> proxyToScheduler(String path, RequestContext rc) {
-    return client.send(rc.request()
-        .withUri(schedulerServiceBaseUrl + SCHEDULER_BASE_PATH + path));
+    return client.send(withRequestId(rc.request()
+        .withUri(schedulerServiceBaseUrl + SCHEDULER_BASE_PATH + path)));
+  }
+
+  private Request withRequestId(Request request) {
+    if (request.headers().containsKey("X-Request-Id")) {
+      return request;
+    }
+    // Unfortunately it is not possible for middleware to set headers on
+    // incoming requests so we manually propagate it here from the MDC.
+    final String requestId = MDC.get("request-id");
+    if (requestId == null) {
```

Comment:
When is this expected to be null?

Reply:
Might be null if the middleware code changes, or a thread pool is introduced without MDC propagation, etc, causing the MDC to not have the request id here. I think it is more important to let the request proxying work instead of failing with a NPE if the request id was unavailable.